src/gui/protocolwidget.cpp

```diff
@@ -183,142 +47,91 @@ ProtocolWidget::ProtocolWidget(QWidget *parent)
+    ConfigFile cfg;
+    cfg.restoreGeometryHeader(_ui->_tableView->horizontalHeader());
+
+    connect(qApp, &QApplication::aboutToQuit, this, [this] {
+        ConfigFile cfg;
+        cfg.saveGeometryHeader(_ui->_tableView->horizontalHeader());
+    });
 
-    QPushButton *copyBtn = _ui->_dialogButtonBox->addButton(tr("Copy"), QDialogButtonBox::ActionRole);
-    copyBtn->setToolTip(tr("Copy the activity list to the clipboard."));
-    copyBtn->setEnabled(true);
-    connect(copyBtn, &QAbstractButton::clicked, this, &ProtocolWidget::copyToClipboard);
+    _ui->_headerLabel->setText(tr("Local sync protocol"));
 }
 
 ProtocolWidget::~ProtocolWidget()
 {
     delete _ui;
 }
 
-void ProtocolWidget::showEvent(QShowEvent *ev)
-{
-    ConfigFile cfg;
-    cfg.restoreGeometryHeader(_ui->_treeWidget->header());
-
-    // Sorting by section was newly enabled. But if we restore the header
-    // from a state where sorting was disabled, both of these flags will be
-    // false and sorting will be impossible!
-    _ui->_treeWidget->header()->setSectionsClickable(true);
-    _ui->_treeWidget->header()->setSortIndicatorShown(true);
-
-    // Switch back to "by time" ordering
-    _ui->_treeWidget->sortByColumn(0, Qt::DescendingOrder);
-
-    QWidget::showEvent(ev);
-}
-
-void ProtocolWidget::hideEvent(QHideEvent *ev)
+void ProtocolWidget::showContextMenu(QWidget *parent, ProtocolItemModel *model, const QModelIndexList &items)
 {
-    ConfigFile cfg;
-    cfg.saveGeometryHeader(_ui->_treeWidget->header());
-    QWidget::hideEvent(ev);
-}
+    auto menu = new QMenu(parent);
+    menu->setAttribute(Qt::WA_DeleteOnClose);
 
-void ProtocolWidget::slotItemContextMenu(const QPoint &pos)
-{
-    auto item = _ui->_treeWidget->itemAt(pos);
-    if (!item)
-        return;
-    auto globalPos = _ui->_treeWidget->viewport()->mapToGlobal(pos);
-    ProtocolItem::openContextMenu(globalPos, item, this);
+    menu->addAction(tr("Copy to clipboard"), parent, [text = Models::formatSelection(items)] {
+        QApplication::clipboard()->setText(text);
+    });
+
+    if (items.size() == 1) {
+        const auto &data = model->protocolItem(items.first());
+        auto folder = FolderMan::instance()->folder(data.folderName());
+        OC_ASSERT(folder);
+        if (!folder)
+            return;
+
+        {
+            const QString localPath = folder->path() + data.path();
+            if (QFileInfo::exists(localPath)) {
+                menu->addAction(tr("Show in file browser"), parent, [localPath] {
+                    if (QFileInfo::exists(localPath)) {
+                        showInFileManager(localPath);
+                    }
+                });
+            }
+        }
+        // "Open in Browser" action
+        {
+            fetchPrivateLinkUrl(folder->accountState()->account(), folder->remotePathTrailingSlash() + data.path(), parent, [parent, menu = QPointer<QMenu>(menu)](const QString &url) {
+                // make sure menu still exists
```

Comment:
Is it okay if the menu does not exist any more then? I think you should include that in your comment.